src/main/java/hudson/plugins/sonar/client/SQProjectResolver.java

```diff
@@ -43,38 +44,34 @@ public SQProjectResolver(HttpClient client) {
    * Errors that should be displayed are included in {@link ProjectInformation#getErrors()}.
    */
   @CheckForNull
-  public ProjectInformation resolve(@Nullable String projectUrl, @Nullable String ceTaskId, String installationName) {
+  public ProjectInformation resolve(@Nullable String serverUrl, @Nullable String projectDashboardUrl, String ceTaskId, String installationName) {
     SonarInstallation inst = SonarInstallation.get(installationName);
     if (inst == null) {
       Logger.LOG.info(() -> "Invalid installation name: " + installationName);
       return null;
     }
 
     try {
-      String projectKey = extractProjectKey(projectUrl);
-      String serverUrl = extractServerUrl(projectUrl);
-
-      if (!checkServerUrl(serverUrl, projectKey, inst)) {
+      if (!checkServerUrl(serverUrl, inst)) {
         return null;
       }
 
       WsClient wsClient = WsClient.create(client, inst);
       Version version = new Version(wsClient.getServerVersion());
 
-      ProjectInformation projectInfo = new ProjectInformation(projectKey);
-      projectInfo.setUrl(projectUrl);
-
-      getQualityGate(wsClient, projectInfo, projectKey, version);
-      if (projectInfo.getStatus() == null) {
-        // if QG is not available for the project, without errors, projectInfo will be empty
-        return projectInfo;
+      if (version.compareTo(new Version("5.6")) < 0) {
+        Logger.LOG.info(() -> "SQ < 5.6 is not supported");
+        return null;
       }
 
-      getCETask(wsClient, projectInfo, ceTaskId);
+      ProjectInformation projectInfo = new ProjectInformation();
+      projectInfo.setUrl(projectDashboardUrl);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Annotate the parameter with @javax.annotation.Nullable in method 'setUrl' declaration, or make sure that null can not be passed as argument. ([squid:S4449](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS4449))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.jenkins-ci.plugins%3Asonar&pullRequest=90&issues=AWIv-I51_wH-bYHl3c0E&open=AWIv-I51_wH-bYHl3c0E)

[ISSUE_KEY]: AWIv-I51_wH-bYHl3c0E